### ISSUE TITLE
Having trouble compiling VICE Libretro and how to put the required library files correctly?

### ISSUE THREAD

**Issue by saulfabregwiivc** [libretro/vice-libretro#511]
Can please anyone tell me how i can install the required libraries used by this Libretro port of VICE so i can compile it properly?

There are too many libraries used by this port and i just can't figure out how i can put the required libraries files into the correct paths required by the files (e.g. on **#include "arch/shared/archdep_access.c"** in **retrodep/archdep.c**, but that file doesn't exist in that exact path but in another path)

Btw, i'm compiling in Windows, but i'm trying to compile a core for the Wii.

Thanks.

**Comment by sonninnos** [comment]
I don't understand what you mean. There are no extra libraries, and everything is included in the repo.

**Comment by saulfabregwiivc** [comment]
I get this error when compiling the resulting dol file:

```
/opt/devkitpro/devkitPPC/bin/powerpc-eabi-g++ -oretroarch_wii.elf -DHW_RVL -mrvl -DGEKKO -mcpu=750 -meabi -mhard-float -Wl,-Map,retroarch_wii.elf.map,-wrap,malloc,-wrap,free,-wrap,memalign,-wrap,calloc,-wrap,realloc,-wrap,strdup,-wrap,strndup,-wrap,malloc_usable_size -T bootstrap/gx/rvl.ld -Lwii/libogc/libs/wii -L. griffin/griffin.o  wii/app_booter/app_booter.binobj   -lretro_wii  -lwiiuse -lbte -logc
.\libretro_wii.a(libretro-glue.o): In function `sevenzip_uncompress':
libretro-glue.c:(.text.sevenzip_uncompress+0x144): undefined reference to `LookToRead2_CreateVTable'
.\libretro_wii.a(archdep.o): In function `archdep_access':
archdep.c:(.text.archdep_access+0x34): undefined reference to `access'
.\libretro_wii.a(archdep.o): In function `archdep_default_resource_file_name':
archdep.c:(.text.archdep_default_resource_file_name+0xac): undefined reference to `access'
archdep.c:(.text.archdep_default_resource_file_name+0xf8): undefined reference to `access'
archdep.c:(.text.archdep_default_resource_file_name+0x1a4): undefined reference to `access'
collect2.exe: error: ld returned 1 exit status
make: *** [makefile.griffin:1247: retroarch_wii.elf] Error 1
```

**Comment by sonninnos** [comment]
`LookToRead2_CreateVTable` is regular 7zip stuff which is under `deps/7zip/`, but Wii uses static linking, which means it isn't using those files. Makefile does have `NO_7ZIP` flag, so you should be able to at least test it without 7zip support.

Regarding that `access` function you might want to copy the one defined for `#ifdef __PS3__`, since it seems common and simple enough:
```c
int access(const char *fpath, int mode)
{
    struct stat buffer;   
    return libretro_stat(fpath, &buffer);
}
```

Then again buildbot is able to build it without that, so something missing in your toolchain then I guess.

**Comment by saulfabregwiivc** [comment]
> `LookToRead2_CreateVTable` is regular 7zip stuff which is under `deps/7zip/`, but Wii uses static linking, which means it isn't using those files. Makefile does have `NO_7ZIP` flag, so you should be able to at least test it without 7zip support.
> 
> Regarding that `access` function you might want to copy the one defined for `#ifdef __PS3__`, since it seems common and simple enough:
> 
> ```c
> int access(const char *fpath, int mode)
> {
>     struct stat buffer;   
>     return libretro_stat(fpath, &buffer);
> }
> ```
> 
> Then again buildbot is able to build it without that, so something missing in your toolchain then I guess.

So i can temporally disable static linking by passing the argument STATIC_LINKING=0 when compiling VICE for Wii, right?

Anyways, i've disabled 7-Zip support here and added the int access as you suggested, but keeping the static linking.
Everything still compiles fine the libretro_wii.a file, now i will see if it now compiles the dol file.

Thanks, i will post more results later.

**Comment by sonninnos** [comment]
No, I don't think you can just disable static linking and expect it to work. It needs RetroArch built-in to be usable. Check the gitlab recipes to see how the template works.
